mhcnames/allele_name.py

```diff
@@ -90,16 +90,20 @@ def parse_allele_name(name, species_prefix=None):
         species = "HLA"
 
     if name[0].upper() == "D":
-        # MHC class II genes like "DQA1" need to be parsed with both
-        # letters and numbers
-        gene, name = parse_alphanum(name, 4)
+        if len(name) == 7:
+            # sometimes we get very compact names like DRB0101
+            gene, name = parse_letters(name, 3)
+        else:
+            # MHC class II genes like "DQA1" need to be parsed with both
+            # letters and numbers
+            gene, name = parse_alphanum(name, 4)
         # TODO: make a list of known species/gene pairs, along with
         # gene synonyms. That should significantly imporve on this kind of
         # ad-hoc synonym handling.
-        if gene == "DRA":
-            gene = "DRA1"
-        elif gene == "DRB":
-            gene = "DRB1"
+
+        if gene.isalpha():
```

Comment:
This is 1 for all the human and NHP alleles. It gets weirder for other species, but that's true of pretty much every change we make to mhcnames, since it never properly handles the edge cases until we explicitly enumerate them.